packages/react/src/primitives/PasswordField/ShowPasswordButton.tsx

```diff
@@ -19,15 +21,25 @@ const ShowPasswordButtonPrimitive: Primitive<
     passwordIsShownLabel = passwordIsShown,
     showPasswordButtonLabel = showPassword,
     size,
+    hasError,
     ...rest
   },
   ref
 ) => {
+  const showPasswordButtonClass = classNames(
+    ComponentClassNames.FieldShowPassword,
+    classNameModifierByFlag(
+      ComponentClassNames.FieldShowPassword,
+      'error',
+      hasError
+    )
+  );
+
```

Comment:
It would be awesome to add a unit tests/assertion to verify props are passed in as expected and error class is applied. There are examples testing these for other components, see [Checkbox.test.tsx](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/src/primitives/Checkbox/__tests__/Checkbox.test.tsx#L70).

Reply:
sure i'll try